cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -188,6 +199,74 @@ public ArtifactClasses inspectArtifact(Id.Artifact artifactId, File artifactFile
   }
 
   /**
+   * load and instantiate the plugin and return {@link PluginEndpoint}
+   * which can be used to invoke plugin method with request object.
+   * @param pluginInstantiator to instantiate plugin instances.
+   * @param artifact artifact of the plugin
+   * @param pluginClass class having the plugin method to invoke
+   * @param endpointName name of the endpoint to invoke
+   * @param endpointPluginContext endpoint plugin context that can optionally be passed to the method
+   * @throws IOException if there was an exception getting the classloader
+   * @throws ClassNotFoundException if plugin class cannot be loaded
+   * @throws NotFoundException Not Found exception thrown if no matching plugin found.
+   */
+  public PluginEndpoint invokePluginEndpoint(final PluginInstantiator pluginInstantiator, Id.Artifact artifact,
```

Comment:
can this be in PluginService instead of here? Doesn't really have anything to do with inspecting artifacts